File: fcfs.c
#include <stdio.h>

void fcfs()
{
    int n, i, j;
    int at[20], bt[20], ct[20], tat[20], wt[20], id[20];
    int current = 0, temp;
    float avgwt = 0, avgtat = 0;

    printf("===== FCFS EMERGENCY QUEUE =====\n");
    printf("Enter number of requests: ");
    scanf("%d", &n);

    for(i=0;i<n;i++)
    {
        id[i] = i + 1;
        printf("\nRequest %d arrival time: ", i+1);
        scanf("%d",&at[i]);
        printf("Request %d burst time: ", i+1);
        scanf("%d",&bt[i]);
    }

    /* Sort by Arrival Time */
    for(i=0;i<n-1;i++)
    {
        for(j=i+1;j<n;j++)
        {
            if(at[i] > at[j])
            {
                temp=at[i]; at[i]=at[j]; at[j]=temp;
                temp=bt[i]; bt[i]=bt[j]; bt[j]=temp;
                temp=id[i]; id[i]=id[j]; id[j]=temp;
            }
        }
    }

    current = 0;
    for(i=0;i<n;i++)
    {
        if(current < at[i])
            current = at[i];

        current += bt[i];

        ct[i] = current;
        tat[i] = ct[i] - at[i];
        wt[i] = tat[i] - bt[i];

        avgwt += wt[i];
        avgtat += tat[i];
    }

    printf("\nREQ\tAT\tBT\tCT\tTAT\tWT\n");
    for(i=0;i<n;i++)
    {
        printf("R%d\t%d\t%d\t%d\t%d\t%d\n",
        id[i],at[i],bt[i],ct[i],tat[i],wt[i]);
    }

    printf("\nGANTT CHART\n\n");

    printf(" ");
    for(i=0;i<n;i++) printf("-------");
    printf("-\n|");

    for(i=0;i<n;i++)
        printf(" R%d |",id[i]);

    printf("\n ");
    for(i=0;i<n;i++) printf("-------");
    printf("-\n");

    current = 0;
    printf("%d",current);

    for(i=0;i<n;i++)
    {
        if(current < at[i])
            current = at[i];

        current += bt[i];
        printf("     %d",current);
    }

    printf("\n\nAverage Turnaround Time = %.2f", avgtat/n);
    printf("\nAverage Waiting Time = %.2f\n", avgwt/n);

}

File: FCFS.java
import java.util.Scanner;

public class FCFS {

    public static void run() {

        Scanner sc = new Scanner(System.in);

        System.out.print(
                "Enter number of processes: ");

        int n = sc.nextInt();

        int[] burst = new int[n];
        int[] waiting = new int[n];
        int[] turnaround = new int[n];

        for (int i = 0; i < n; i++) {

            System.out.print(
                    "Burst time P" + (i + 1) + ": ");

            burst[i] = sc.nextInt();
        }

        waiting[0] = 0;

        for (int i = 1; i < n; i++) {

            waiting[i] =
                    waiting[i - 1] + burst[i - 1];
        }

        for (int i = 0; i < n; i++) {

            turnaround[i] =
                    waiting[i] + burst[i];
        }

        double avgWaiting = 0;
        double avgTurnaround = 0;

        System.out.println(
                "\nProcess\tBurst\tWaiting\tTurnaround");

        for (int i = 0; i < n; i++) {

            System.out.println(
                    "P" + (i + 1)
                    + "\t"
                    + burst[i]
                    + "\t"
                    + waiting[i]
                    + "\t"
                    + turnaround[i]);

            avgWaiting += waiting[i];
            avgTurnaround += turnaround[i];
        }

        System.out.println(
                "\nAverage Waiting Time = "
                + avgWaiting / n);

        System.out.println(
                "Average Turnaround Time = "
                + avgTurnaround / n);
    }
}